tests/arp/test_tagged_arp.py

```diff
@@ -0,0 +1,259 @@
+    config_port_channel_member_ports = list(itertools.chain.from_iterable(config_port_channel_members))
+    config_ports_vlan = defaultdict(list)
+    vlan_members = cfg_facts.get('VLAN_MEMBER', {})
+    # key is dev name, value is list for configured VLAN member.
+    for k, v in cfg_facts['VLAN'].items():
+        vlanid = v['vlanid']
+        for addr in cfg_facts['VLAN_INTERFACE']['Vlan'+vlanid]:
+            # address could be IPV6 and IPV4, only need IPV4 here
+            if addr.find(':') == -1:
+                ip = addr
+                break
+        else:
+            continue
+        for port in v['members']:
+            if k in vlan_members and port in vlan_members[k]:
+                if 'tagging_mode' not in vlan_members[k][port]:
+                    continue
+                mode = vlan_members[k][port]['tagging_mode']
+                config_ports_vlan[port].append({'vlanid':int(vlanid), 'ip':ip, 'tagging_mode':mode})
+
+    if config_portchannels:
+        portchannel_cnt = 0
+        for po in config_portchannels:
+            vlan_port = {
+                'dev' : po,
+                'port_index' : [config_port_indices[member] for member in config_portchannels[po]['members']],
+                'permit_vlanid' : []
+            }
+            if po in config_ports_vlan:
+                vlan_port['pvid'] = 0
+                for vlan in config_ports_vlan[po]:
+                    if 'vlanid' not in vlan or 'ip' not in vlan or 'tagging_mode' not in vlan:
+                        continue
+                    if vlan['tagging_mode'] == 'untagged':
+                        vlan_port['pvid'] = vlan['vlanid']
+                    vlan_port['permit_vlanid'].append(vlan['vlanid'])
+            if 'pvid' in vlan_port:
+                vlan_ports_list.append(vlan_port)
+
+    ports = [port for port in config_ports
+        if config_port_indices[port] in ptf_ports_available_in_topo
+        and config_ports[port].get('admin_status', 'down') == 'up'
+        and port not in config_port_channel_member_ports]
+
+    for i, port in enumerate(ports):
+        vlan_port = {
+            'dev' : port,
+            'port_index' : [config_port_indices[port]],
+            'permit_vlanid' : []
+        }
+        if port in config_ports_vlan:
+            vlan_port['pvid'] = 0
+            for vlan in config_ports_vlan[port]:
+                if 'vlanid' not in vlan or 'ip' not in vlan or 'tagging_mode' not in vlan:
+                    continue
+                if vlan['tagging_mode'] == 'untagged':
+                    vlan_port['pvid'] = vlan['vlanid']
+                vlan_port['permit_vlanid'].append(vlan['vlanid'])
+        if 'pvid' in vlan_port:
+            vlan_ports_list.append(vlan_port)
+
+    return vlan_ports_list
+
+
+def enable_arp(duthost, cfg_facts, enable):
+    vlan_members = cfg_facts.get('VLAN_MEMBER', {})
+    on_cmd = "echo 1 > /proc/sys/net/ipv4/conf/%s/arp_accept"
```

Comment:
>echo 1 > /proc/sys/net/ipv4/conf/%s/arp_accept [](http://example.com/codeflow?start=14&length=46)

Is it really necessary? What is the default behavior of sonic image?

Reply:
Test again, still need to enable arp_accept.

Reply:
arp_accept is for garp. Why don't the test send regular arp request packet here?

Reply:
> arp_accept is for garp. Why don't the test send regular arp request packet here?

I have verified arp request packet, but still got the same result.